cmd/snap/cmd_get_base_declaration_test.go

```diff
@@ -0,0 +1,55 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"gopkg.in/check.v1"
+
+	snap "github.com/snapcore/snapd/cmd/snap"
+)
+
+func (s *SnapSuite) TestGetBaseDeclaration(c *check.C) {
+	n := 0
+	s.RedirectClientToTestServer(func(w http.ResponseWriter, r *http.Request) {
+		switch n {
+		case 0:
+			c.Check(r.Method, check.Equals, "POST")
+			c.Check(r.URL.Path, check.Equals, "/v2/debug")
+			c.Check(r.URL.RawQuery, check.Equals, "")
+			data, err := ioutil.ReadAll(r.Body)
+			c.Check(err, check.IsNil)
+			c.Check(data, check.DeepEquals, []byte(`{"action":"get-base-declaration"}`))
+			fmt.Fprintln(w, `{"type": "sync", "result": "aGVsbG8K"}`)
+		default:
+			c.Fatalf("expected to get 1 requests, now on %d", n+1)
+		}
+
+		n++
+	})
+	rest, err := snap.Parser().ParseArgs([]string{"debug", "get-base-declaration"})
+	c.Assert(err, check.IsNil)
+	c.Assert(rest, check.DeepEquals, []string{})
+	c.Check(s.Stdout(), check.Equals, "hello\n")
```

Comment:
Sorry for being slow, but where does this "hello" come from? I see above that there is a  `"result": "aGVsbG8K"` but no "hello"?

While at it, I think I would prefer something slightly closer to the real output from the daemon, i.e. if someone looks at this test ideally he/she would get a feel what kind of data to expect from the daemon.